src/mame/drivers/meritum.cpp

```diff
@@ -418,9 +493,13 @@ void meritum_state::meritum(machine_config &config)
 	pit.out_handler<2>().set_inputline(m_maincpu, INPUT_LINE_NMI);
 
 	i8255_device &mainppi(I8255(config, "mainppi")); // parallel interface
-	mainppi.out_pc_callback().set("nmigate", FUNC(input_merger_device::in_w<0>)).bit(7).invert();
+	mainppi.out_pc_callback().set(FUNC(meritum_state::mainppi_portc_w));
+	mainppi.out_pb_callback().set(FUNC(meritum_state::mainppi_portb_w));
 
-	I8255(config, "flopppi", 0); // floppy disk interface
+	if( m_model > 1)
+	{
+		I8255(config, "flopppi", 0); // floppy disk interface
+	}
```

Comment:
Please make the model 2 machine configuration function call the model 1 machine configuration function then add this.